test/server/cards/02_SHD/upgrades/InfiltratorsSkill.spec.js

```diff
@@ -0,0 +1,34 @@
+describe('Infiltrator\'s Skill', function() {
+    integration(function() {
+        describe('Infiltrator\'s Skill\'s ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [{ card: 'battlefield-marine', upgrades: ['infiltrators-skill'] }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'pyke-sentinel', upgrades: ['shield'] }]
+                    }
+                });
+            });
+
+            it('should ignore sentinel', function () {
+                this.player1.clickCard(this.battlefieldMarine);
+                expect(this.player1).toBeAbleToSelectExactly([this.pykeSentinel, this.p2Base]);
+                // ignore sentinel
+                this.player1.clickCard(this.p2Base);
+                expect(this.p2Base.damage).toBe(4);
+            });
+            it('should ignore shield', function () {
+                this.player1.clickCard(this.battlefieldMarine);
+                expect(this.player1).toBeAbleToSelectExactly([this.pykeSentinel, this.p2Base]);
+
+                // defeat shielded pyke sentinel
+                this.player1.clickCard(this.pykeSentinel);
+                expect(this.battlefieldMarine.damage).toBe(2);
+                expect(this.pykeSentinel).toBeInLocation('discard');
```

Comment:
Can we change this test to use a unit that will survive the attack? We want the most precise test possible, since some effect could defeat the unit and shield together (though unlikely).

So the best test would be to have a bigger unit and see that (a) its shield is defeated and (b) it has taken damage from the attack, meaning that the shield was defeated before the attack damage happened.